src/main/scala/org/broadinstitute/hail/driver/BuildDocs.scala

```diff
@@ -0,0 +1,89 @@
+package org.broadinstitute.hail.driver
+
+import org.broadinstitute.hail.Utils._
+import org.kohsuke.args4j.{Option => Args4jOption, CmdLineParser}
+import scala.collection.JavaConverters._
+import org.json4s._
+import org.json4s.jackson.JsonMethods._
+
+object BuildDocs extends Command {
+
+  class Options extends BaseOptions {
+    @Args4jOption(required = true, name = "-o", aliases = Array("--output"), usage = "output file location")
+    var outputFile: String = _
+  }
+
+  def description = "Builds documentation"
+
+  def name = "builddocs"
+
+  override def hidden = true
+
+  def requiresVDS = false
+
+  def supportsMultiallelic = true
+
+  def newOptions = new Options
+
+  def run(state: State, options: Options): State = {
+
+    val orderedCommandList = ToplevelCommands.commands.flatMap {
+      case (s, y: SuperCommand) => Option(y.subcommands.map { case (a, b) => (s + " " + a, b) })
+      case (s, x: Command) => Option(Map[String, Command]((s, x)))
+      case _ => None
+    }.flatten.toIndexedSeq.sortWith((x1, x2) => x1._1 < x2._1)
+
+    def createSynopsis(cmdName: String, parser: CmdLineParser): String = {
```

Comment:
I'm getting into the weeds here, but I like just `synopsis` rather than `createSynopsis`.  It is natural to let a function name describe its return value (esp. when it is functional).